src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -301,6 +302,12 @@ create_reference_table(PG_FUNCTION_ARGS)
 	 */
 	EnsureRelationKindSupported(relationId);
 
+	/*
+	 * Take an exclusive lock on pg_dist_node to serialize pg_dist_node changes
+	 * with reference table distribution
+	 */
+	LockRelationOid(DistNodeRelationId(), ShareLock);
```

Comment:
Multiple reference tables should be able to be created concurrently, so I'd use `RowExclusiveLock`?

Reply:
ShareLock might make sense considering that it blocks all writes, but not itself.  Comment is stale.

Reply:
I've updated it to `AccessShareLock` because our use is only reading from the list